File: app/src/types/__tests__/SupplyStatus.test.ts
import { describe, it, expect } from "vitest";
import { SupplyStatus } from "../SupplyStatus";

describe("SupplyStatus", () => {
  it("should have three enum values", () => {
    expect(SupplyStatus.SUFFICIENT).toBe("SUFFICIENT");
    expect(SupplyStatus.INSUFFICIENT).toBe("INSUFFICIENT");
    expect(SupplyStatus.MISSING).toBe("MISSING");
  });

  it("should allow enum value comparison", () => {
    const status = SupplyStatus.SUFFICIENT;
    expect(status).toBe(SupplyStatus.SUFFICIENT);
    expect(status).not.toBe(SupplyStatus.MISSING);
  });
});

File: HSAPI/src/test/kotlin/com/hobbystock/types/SupplyStatusTest.kt
package com.hobbystock.types

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SupplyStatusTest {
    @Test
    fun `should have three status values`() {
        val values = SupplyStatus.values()
        assertThat(values).hasSize(3)
        assertThat(values)
                .containsExactlyInAnyOrder(
                        SupplyStatus.SUFFICIENT,
                        SupplyStatus.INSUFFICIENT,
                        SupplyStatus.MISSING
                )
    }
}
